src/main/java/com/diffplug/gradle/eclipse/apt/AptPlugin.java

```diff
@@ -109,15 +109,14 @@ private <T extends AbstractCompile> void configureCompileTasks(
       final Project project,
       Class<T> compileTaskClass,
       final Function<T, CompileOptions> getCompileOptions) {
-    IMPL.configureTasks(
-        project,
-        compileTaskClass,
-        task -> {
-          CompileOptions compileOptions = getCompileOptions.apply(task);
-          final AptOptions aptOptions = IMPL.createAptOptions();
-          task.getExtensions().add(AptOptions.class, "aptOptions", aptOptions);
-          IMPL.configureCompileTask(task, compileOptions, aptOptions);
-        });
+    project.afterEvaluate(p -> {
+        for (T task : p.getTasks().withType(compileTaskClass)) {
+            CompileOptions compileOptions = getCompileOptions.apply(task);
+            final AptOptions aptOptions = IMPL.createAptOptions();
+            task.getExtensions().add(AptOptions.class, "aptOptions", aptOptions);
+            IMPL.configureCompileTask(task, compileOptions, aptOptions);
+        }
+    });
```

Comment:
While Gradle's own `eclipseClasspath` task is not currently compatible with the configuration cache, adding to task extensions using this configuration block is disallowed because it could be resolved at execution time. Adding to these tasks' extensions in `Project#afterEvaluate` guarantees that this doesn't happen, and that your extensions to `eclipseClasspath` will not hinder the performance of Gradle's task once they update it.